src/fondant/pipeline/pipeline.py

```diff
@@ -319,11 +328,136 @@ def __init__(
         self._graph: t.OrderedDict[str, t.Any] = OrderedDict()
         self.task_without_dependencies_added = False
 
+    def read(self,
+             name,
+             *,
+             arguments: t.Optional[t.Dict[str, t.Any]] = None,
+             input_partition_rows: t.Optional[t.Union[str, int]] = None,
+             cache: t.Optional[bool] = True,
+             cluster_type: t.Optional[str] = "default",
+             client_kwargs: t.Optional[dict] = None,
+             resources: t.Optional[Resources] = None,
+             schema: t.Optional[t.Dict[str, str]] = None) -> "Pipeline":
+
+        component_op = self._build_component_op(name,
+                                                arguments=arguments,
+                                                input_partition_rows=input_partition_rows,
+                                                cache=cache,
+                                                cluster_type=cluster_type,
+                                                client_kwargs=client_kwargs,
+                                                resources=resources,
+                                                schema=schema)
+
+        self.add_op(component_op)
+        return self
```

Comment:
General question 

Do we need to return the class instance. So instead of this:

```python
pipeline = Pipeline(
    pipeline_name="my_pipeline",
    pipeline_description="description of my pipeline",
    base_path="/foo/bar",
)

dataset = pipeline.read(
    name="load_images",
    schema={
        "image": type.binary  # or pa.binary()
    }
)
```

We would have: 

```python
pipeline = Pipeline(
    pipeline_name="my_pipeline",
    pipeline_description="description of my pipeline",
    base_path="/foo/bar",
)

pipeline.read(
    name="load_images",
    schema={
        "image": type.binary  # or pa.binary()
    }
)
```

A bit similar to the the way we used to add operations to the pipeline. I think both are valid but not sure which one is more intuitive

Reply:
The idea behind returning the class instance was that to call apply on a `dataset` afterwards.
Returning the class instance gives the user the most flexibility. The user can decide which variant he wants to implement. 

Either:
```python
dataset = pipeline.read(...)
dataset = dataset.apply(...)
```

or 

```python
pipeline.read(...)
dataset.apply(...)
```

or even: 

```python
pipeline.read(...).apply(...).apply(...)
```

Reply:
ok that makes sense :)

Reply:
Also, the `Dataset` class can provide an interface for more interactive data exploration.